Added button to copy file name in PR files

saneks222 opened the pull request:
The merge request file viewer has a button for copying the file path, but it is not always convenient. Often, you only want to copy the file name, which is currently not possible. This change request adds this capability.

saneks222 commented:
> It seems to be a duplicate button.

Thank you for your reply!
If you think this button is redundant here, perhaps we could move it here? What do you think about this option? 

<img width="1903" height="543" alt="image" src="https://github.com/user-attachments/assets/807a19f0-46ca-4acc-b731-02695f3f14fd" />

saneks222 commented:
> Call it "Copy filename" and have it in that menu, yes.

I moved the button to the menu and also attempted to add localization, but encountered issues with Chinese dialects, so I was unable to add localization for it

wxiaoguang commented:
> > Call it "Copy filename" and have it in that menu, yes.
> 
> I moved the button to the menu and also attempted to add localization, but encountered issues with Chinese dialects, so I was unable to add localization for it

No need to make other translations. The only thing need to do is to add the en-US. Other languages will be handled by Crowdin.

By the way, there could be some improvments:

1. No need to introduce `ShortName`. You can call `data-xxx="{{$file.GetDiffFileShortName}}"` directly from template
2. It should use `path.Base` but not `filepath.Base`

saneks222 commented:
> > > Call it "Copy filename" and have it in that menu, yes.
> > 
> > 
> > I moved the button to the menu and also attempted to add localization, but encountered issues with Chinese dialects, so I was unable to add localization for it
> 
> No need to make other translations. The only thing need to do is to add the en-US. Other languages will be handled by Crowdin.
> 
> By the way, there could be some improvments:
> 
> 1. No need to introduce `ShortName`. You can call `data-xxx="{{$file.GetDiffFileShortName}}"` directly from template
> 2. It should use `path.Base` but not `filepath.Base`

Thanks for the review!
Fixed.

wxiaoguang commented on templates/repo/diff/box.tmpl:
```diff
@@ -143,6 +143,7 @@
 											<button class="unescape-button item" data-unicode-content-selector="#diff-{{$file.NameHash}}">{{ctx.Locale.Tr "repo.unescape_control_characters"}}</button>
 											<button class="escape-button tw-hidden item" data-unicode-content-selector="#diff-{{$file.NameHash}}">{{ctx.Locale.Tr "repo.escape_control_characters"}}</button>
 										{{end}}
+										<button class="copy-filename-button item" data-clipboard-text="{{$file.GetDiffFileShortName}}">{{ctx.Locale.Tr "copy_filename"}}</button>
```
```suggestion
										<button class="item" data-clipboard-text="{{$file.GetDiffFileShortName}}">{{ctx.Locale.Tr "copy_filename"}}</button>
```

wxiaoguang commented on services/gitdiff/gitdiff.go:
```diff
@@ -401,6 +402,14 @@ func (diffFile *DiffFile) GetDiffFileName() string {
 	return diffFile.Name
 }
 
+// GetDiffFileName returns the short name of the diff file, or its short old name in case it was deleted
+func (diffFile *DiffFile) GetDiffFileShortName() string {
```
```suggestion
// GetDiffFileBaseName returns the short name of the diff file, or its short old name in case it was deleted
func (diffFile *DiffFile) GetDiffFileBaseName() string {
```

wxiaoguang commented on templates/repo/diff/box.tmpl:
```diff
@@ -143,6 +143,7 @@
 											<button class="unescape-button item" data-unicode-content-selector="#diff-{{$file.NameHash}}">{{ctx.Locale.Tr "repo.unescape_control_characters"}}</button>
 											<button class="escape-button tw-hidden item" data-unicode-content-selector="#diff-{{$file.NameHash}}">{{ctx.Locale.Tr "repo.escape_control_characters"}}</button>
 										{{end}}
+										<button class="item" data-clipboard-text="{{$file.GetDiffFileShortName}}">{{ctx.Locale.Tr "copy_filename"}}</button>
```
```suggestion
										<button class="item" data-clipboard-text="{{$file.GetDiffFileBaseName}}">{{ctx.Locale.Tr "copy_filename"}}</button>
```
